src/WrapCalls.cpp

```diff
@@ -0,0 +1,114 @@
+#include "WrapCalls.h"
+#include "IRVisitor.h"
+
+#include <set>
+
+namespace Halide{
+namespace Internal {
+
+using std::map;
+using std::pair;
+using std::set;
+using std::string;
+using std::vector;
+
+void wrap_func_calls_helper(map<string, Function> &env, const string &in_func,
+                            const Function &orig, const Function &substitute) {
+    env[in_func] = env[in_func].substitute_calls(orig, substitute);
+}
+
+// Return true if 'func' exists as a copy of one of the function in 'copied_map'
+bool is_copy(const Function& func, const map<Function, Function> copied_map) {
```

Comment:
Should make this copied_map param a const reference to avoid a copy constructed map.